test/framework/resource/k8s/node/manager.go

```diff
@@ -95,3 +99,21 @@ func (d *defaultManager) GetNode(node *v1.Node) (*v1.Node, error) {
 	err := d.k8sClient.Get(context.TODO(), utils.NamespacedName(node), observedNode)
 	return observedNode, err
 }
+
+func (d *defaultManager) GetCNINode(node *v1.Node) (*cninode.CNINode, error) {
```

Comment:
I wonder how many of these functions you could remove with golang generics :).

Reply:
Yeah, I can explore them. 👍